text/001x-battery_improvements.md

```diff
@@ -0,0 +1,293 @@
+      </entry>
+      <entry value="65536" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_FIRMWARE">
+        <description>Battery firmware is not compatible with current autopilot firmware.</description>
+      </entry>
+      <entry value="131072" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_CELLS_CONFIGURATION">
+        <description>Battery is not compatible due to cell configuration (e.g. 5s1p when vehicle requires 6s).</description>
+      </entry>
+      <entry value="262144" name="MAV_BATTERY_STATUS_FLAGS_CAPACITY_RELATIVE_TO_FULL">
+        <description>
+          Battery capacity values are relative to a known-full battery.
+          If set: the supplied capacity_consumed and capacity_remaining are relative to battery full.
+          If unset: capacity_consumed is relative to drone power-on, and capacity_remaining estimated from capacity_consumed and the assumption that the battery was full at power on.
+          This bit should be set by smart batteries and unset for power modules.
+          If unset a GCS is recommended to advise that users fully charge the battery on power on.
+        </description>
+      </entry>
+      <entry value="4,294,967,295" name="MAV_BATTERY_STATUS_FLAGS_EXTENDED">
+        <description>Reserved (not used). If set, this will indicate that an additional status field exists for higher status values.</description>
+      </entry>
+    </enum>
+```
+
+The message is heavily based on [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) but:
+- removes the cell voltage arrays: `voltages` and `voltages_ext`
+- adds `voltage`, the total cell voltage. This is a uint32_t to allow batteries with more than 65V.
+- `current_consumed` has changed to capacity_consumed, and from an `int32_t` to a `uint32_t`.
+  This future proofs from 32767 mAh limit to 4294967295 mAh 
+- removes `energy_consumed`, which essentially duplicates `capacity_consumed` for most purposes, and `capacity_consumed` in mAh is nearly ubiquitous.
+- adds `capacity_remaining`(mAh).
+  - This allows a GCS to more accurately determine available current and remaining time than inferring from the `capacity_consumed` and `percent_remaining`.
+  - This will be reliable for smart batteries.
+  - A flag has been added to indicate whether this is calculated from an assumed or known-full battery.
+    A GCS can use this to prompt that batteries be fully charged for power modules.
+- Note that with capacity consumed and remaining, you have the full capacity and you could calculate the percentage remaining.
+  However percentage remaining is supplied anyway, as the other values are optional, and this is actually the one value that most users really want.
+- change `current` from a `int16_t` (cA) to a `uint32_t` (mA). Maximum size was previously 327.67A, which is not large enough to be future proof. The value is absolute - if you're charging the value can be assumed to be in a reversed direction.
```

Comment:
This may require some gcs changes to check the charging flag vs. just using int32 and not "needing" to do that though you should ?

Reply:
As above, yes, let's fix this to an int32. Plenty big enough.

Reply:
```suggestion
- change `current` from a `int16_t` (cA) to a `int32_t` (mA). Maximum size was previously 327.67A, which is not large enough to be future proof. New value gives up to 2,147,483A. The value is positive when discharging, and negative when charging.
```

Reply:
great!